docs/data-analysis.md

```diff
@@ -1,164 +1,108 @@
-                            ticker_interval = ticker_interval,
-                            pair=pair)
-print(len(bt_data))
-
-### Start strategy reload
-# Load strategy - best done in a new cell
-# Rerun each time the strategy-file is changed.
-strategy = StrategyResolver({'strategy': strategyname,
-                            'user_data_dir': Path.cwd(),
-                            'strategy_path': location}).strategy
+# Show value-counts per pair
+df.groupby("pair")["sell_reason"].value_counts()
+```
 
-# Run strategy (just like in backtesting)
-df = strategy.analyze_ticker(bt_data, {'pair': pair})
-print(f"Generated {df['buy'].sum()} buy signals")
+### Load live trading results into a pandas dataframe
 
-# Reindex data to be "nicer" and show data
-data = df.set_index('date', drop=True)
-data.tail()
+``` python
+# Fetch trades from database
+df = load_trades_from_db("sqlite:///tradesv3.sqlite")
 
+# Display results
+df.groupby("pair")["sell_reason"].value_counts()
 ```
 
-### Explanation
+## Strategy debugging example
 
-#### Imports and constant definition
+Debugging a strategy can be time-consuming. FreqTrade offers helper functions to visualize raw data.
 
-``` python
-# Some necessary imports
-from pathlib import Path
+### Import requirements and define variables used in analyses
 
+```python
+# Imports
+from pathlib import Path
+import os
 from freqtrade.data.history import load_pair_history
 from freqtrade.resolvers import StrategyResolver
-# Define some constants
-ticker_interval = "5m"
+from freqtrade.data.btanalysis import load_backtest_data
+from freqtrade.data.btanalysis import load_trades_from_db
 
+# Define some constants
+ticker_interval = "1m"
 # Name of the strategy class
-strategyname = 'Awesomestrategy'
+strategy_name = 'NewStrategy'
+# Path to user data
+user_data_dir = 'user_data'
 # Location of the strategy
-strategy_location = 'user_data/strategies'
+strategy_location = Path(user_data_dir, 'strategies')
 # Location of the data
-data_location = 'user_data/data/binance'
+data_location = Path(user_data_dir, 'data', 'binance')
+# Pair to analyze 
 # Only use one pair here
-pair = "XRP_ETH"
+pair = "BTC_USDT"
 ```
 
-This first section imports necessary modules, and defines some constants you'll probably need to adjust for your case.
+### Load exchange data
 
-#### Load candles
-
-``` python
-# Load data
+```python
+# Load data using values set above
 bt_data = load_pair_history(datadir=Path(data_location),
-                            ticker_interval = ticker_interval,
+                            ticker_interval=ticker_interval,
                             pair=pair)
-print(len(bt_data))
-```
 
-This second section loads the historic data and prints the amount of candles in the DataFrame.
-You can also inspect this dataframe by using `bt_data.head()` or `bt_data.tail()`.
+# Confirm success
+print("Loaded " + str(len(bt_data)) + f" rows of data for {pair} from {data_location}")
```

Comment:
this is a very odd mixture of using f-strings and don't using them - i would prefer to be everything a f-string (which will also remove the need for "str()" ...

Reply:
Indeed it is. I cleaned it up.